src/llvm-late-gc-lowering.cpp

```diff
@@ -2484,16 +2484,37 @@ bool LateLowerGCFrame::CleanupIR(Function &F, State *S, bool *CFGModified) {
                 NewCall->copyMetadata(*CI);
                 CI->replaceAllUsesWith(NewCall);
                 UpdatePtrNumbering(CI, NewCall, S);
-            } else if (CI->arg_size() == CI->getNumOperands()) {
-                /* No operand bundle to lower */
-                ++it;
-                continue;
             } else {
-                CallInst *NewCall = CallInst::Create(CI, None, CI);
-                NewCall->takeName(CI);
-                NewCall->copyMetadata(*CI);
-                CI->replaceAllUsesWith(NewCall);
-                UpdatePtrNumbering(CI, NewCall, S);
+                if (CI->getFnAttr("julia.gc_safe").isValid()) {
+                    // Insert the operations to switch to gc_safe if necessary.
+                    IRBuilder<> builder(CI);
+                    Value *pgcstack = getPGCstack(F);
+                    assert(pgcstack);
+                    Value *ptls = get_current_ptls_from_task(builder, T_size, get_current_task_from_pgcstack(builder, T_size, pgcstack), tbaa_gcframe);
+                    Value *last_gc_state = emit_gc_safe_enter(builder, T_size, ptls, false);
+                    builder.SetInsertPoint(CI->getNextNode());
+                    // Can't use `emit_gc_safe_exit` since that wan'ts to emit some branches...
```

Comment:
I am guessing you would need to make a new block for this specific emit call, then fix up the IR by calling [SplitBlockAndInsertIfThenElse](https://llvm.org/doxygen/namespacellvm.html#a7ac00229d8c59902686f52ed061cdc80) and splice the instructions back into their respective parts of the IfThenElse blocks and delete the new blocks

Reply:
Yeah I was trying that and had an infinite loop in the iteration here so I am not sure if this is the right strategy